src/internal/utils/sort.ts

```diff
@@ -64,16 +64,25 @@ export function getAttributeSortItem(
     return attributeSortItem;
 }
 
-export function getFirstAttributeIdentifier(
-    resultSpec: AFM.IResultSpec,
-    dimensionIndex: number,
-): string | null {
+function ignoreMeasureGroup(item: string): boolean {
+    return item !== MEASUREGROUP;
+}
+
+function getDimensionItems(resultSpec: AFM.IResultSpec, dimensionIndex: number): AFM.Identifier[] {
     const dimensionItems: AFM.Identifier[] = get(
         resultSpec,
         ["dimensions", dimensionIndex, "itemIdentifiers"],
         [],
     );
-    return dimensionItems.find(a => a !== MEASUREGROUP) || null;
+    return dimensionItems.filter(ignoreMeasureGroup) || [];
+}
+
+export function getFirstAttributeIdentifier(
+    resultSpec: AFM.IResultSpec,
+    dimensionIndex: number,
+): string | null {
+    const dimensionItems = getDimensionItems(resultSpec, dimensionIndex);
+    return dimensionItems.find(ignoreMeasureGroup) || null;
```

Comment:
`dimensionItems` already filtered and all items is attribute.
Just get first items.

Reply:
fixed